web/app/js/components/util/TapUtils.jsx

```diff
@@ -78,10 +78,18 @@ export const processNeighborData = (source, labels, resourceAgg, resourceType) =
     };
   }
 
+  // keep track of pods under this resource to display the number of unmeshed source pods
+  neighb.pods = {};
+  !labels.pod ? null : neighb.pods[labels.pod] = true;
+
   let key = neighb.type + "/" + neighb.name;
   if (_.has(labels, "control_plane_ns")) {
     delete resourceAgg[key];
   } else {
+    if (_.has(resourceAgg, key)) {
+      neighb.pods = resourceAgg[key].pods;
+      !labels.pod ? null : neighb.pods[labels.pod] = true;
+    }
```

Comment:
I had a bit of trouble reasoning thorough this. Seeing the same ternary in two places was a bit confusing, plus I don't tend to use ternary expressions for side effects. Maybe this is a bit easier to follow?

```diff
diff --git a/web/app/js/components/util/TapUtils.jsx b/web/app/js/components/util/TapUtils.jsx
index aa34c168..704b23d7 100644
--- a/web/app/js/components/util/TapUtils.jsx
+++ b/web/app/js/components/util/TapUtils.jsx
@@ -80,15 +80,16 @@ export const processNeighborData = (source, labels, resourceAgg, resourceType) =
 
   // keep track of pods under this resource to display the number of unmeshed source pods
   neighb.pods = {};
-  !labels.pod ? null : neighb.pods[labels.pod] = true;
+  if (labels.pod) {
+    neighb.pods[labels.pod] = true;
+  }
 
   let key = neighb.type + "/" + neighb.name;
   if (_.has(labels, "control_plane_ns")) {
     delete resourceAgg[key];
   } else {
     if (_.has(resourceAgg, key)) {
-      neighb.pods = resourceAgg[key].pods;
-      !labels.pod ? null : neighb.pods[labels.pod] = true;
+      neighb.pods = _.merge(neighb.pods, resourceAgg[key].pods);
     }
     resourceAgg[key] = neighb;
   }
```

Reply:
ah, sounds reasonable! changed.